crates/bevy_pbr/src/render/pbr_functions.wgsl

```diff
@@ -268,3 +269,16 @@ fn dither(color: vec4<f32>, pos: vec2<f32>) -> vec4<f32> {
 }
 #endif
 
+fn premultiply_alpha(standard_material_flags: u32, color: vec4<f32>) -> vec4<f32> {
+    let alpha_mode = standard_material_flags & STANDARD_MATERIAL_FLAGS_ALPHA_MODE_BITS;
+    if (alpha_mode == STANDARD_MATERIAL_FLAGS_ALPHA_MODE_BLEND) {
```

Comment:
would using a switch statement be better here? That's a lot of conditional checks it would need to do to get to the last branch and I have no idea if the compiler is smart enough to optimize it.

Reply:
Looks like switches in WGSL only allow literals:

<img width="801" alt="image" src="https://user-images.githubusercontent.com/418473/202330992-92ca87d2-a37a-444d-99bb-fb135130d198.png">

I could add comments (or maybe defines instead of constants?) but maybe it would be less readable? Let me know what you prefer

Reply:
Using an additional shader def, I cleaned it up so that only two conditionals are needed now. Hopefully it will optimize better.